src/tunacode/tools/bash.py

```
@@ -132,44 +131,44 @@ def _validate_command_security(command: str) -> None:
-            raise ModelRetry(f"Potentially unsafe pattern detected in command: {pattern}")
+            raise ToolRetryError(f"Potentially unsafe pattern detected in command: {pattern}")
 
     # Check for restricted characters (but allow safe environment variable usage)
     # Allow $ when used for legitimate environment variables or shell variables
     if re.search(r"\$[^({a-zA-Z_]", command):
         # $ followed by something that's not a valid variable start
-        raise ModelRetry("Potentially unsafe character '$' in command")
+        raise ToolRetryError("Potentially unsafe character '$' in command")
 
     # Check other restricted characters but allow { } when part of valid variable expansion
     if "{" in command or "}" in command:  # noqa: SIM102
```

Ruff (0.14.14)

Unused `noqa` directive (non-enabled: `SIM102`)

Remove unused `noqa` directive

(RUF100)